include/dlaf/eigensolver/bt_band_to_tridiag/mc.h

```diff
@@ -135,8 +138,131 @@ auto updateE(pika::threads::thread_priority priority, WSender&& tile_w, W2Sender
+    // 0 0 0 d      Y Y Y Y
+    //
+    // From the drawing above, it is possible to see the dashed tile separation between X and Y,
+    // and how the reflectors on the left are going to be applied. In particular, the first row of
+    // the upper tile is not affected.
+
+    acrossTiles_ = [&]() {
+      // Note:
+      // A single tile is involved if:
+      // - it is the last row tile, so by construction reflectors will be applied to a single tile;
+      // - applying reflectors falls entirely withing a single tile
+      const SizeType i_tile = dist.globalTileFromGlobalElement<Coord::Row>(offset.row());
+      const bool is_last_row_tile = i_tile == dist.nrTiles().rows() - 1;
+      return !(is_last_row_tile || sub_offset.row() + 1 + fullsize <= mb);
+    }();
+
+    if (acrossTiles_) {
+      const auto part0_nrows = b - 1;
+      parts_[0] = part_t(sub_offset.row() + 1, part0_nrows, nrefls);
+      parts_[1] = part_t(0, std::min(b, rows_below - b), nrefls, part0_nrows);
+    }
+    else {
+      parts_[0] = part_t(sub_offset.row() + 1, std::min(fullsize, rows_below - 1), nrefls);
+    }
+  };
+
+  // Return true if the application of Householder reflectors involves multiple tiles
+  bool affectsMultipleTiles() const noexcept {
+    return acrossTiles_;
+  }
+
+  // Return SubTileSpec to use for accessing Householder reflectors in compact form
+  matrix::SubTileSpec specHHCompact() const noexcept {
+    return input_spec_;
+  }
+
+  // Return SubTileSpec to use for accessing Householder reflectors in well formed form
+  matrix::SubTileSpec specHH() const noexcept {
+    return {{0, 0}, {parts_[0].rows() + parts_[1].rows(), nrefls_}};
+  }
+
+  // Return requested part, which gives access to specs (see part_t for more info)
+  const auto& operator[](const SizeType index) const noexcept {
+    DLAF_ASSERT_MODERATE(index >= 0 && index < (affectsMultipleTiles() ? 2 : 1), index,
+                         affectsMultipleTiles());
+    return parts_[to_sizet(index)];
+  }
+
+private:
+  struct part_t {
+    // Create an "empty" part
+    part_t() = default;
+
+    part_t(const SizeType origin, const SizeType nrows, const SizeType nrefls, const SizeType offset = 0)
+        : origin_(origin), nrows_(nrows), nrefls_(nrefls), offset_(offset) {}
+
+    // Return the number of rows that this part involves
+    SizeType rows() const noexcept {
+      return nrows_;
+    }
+
+    // Return the spec to use on well-formed HH
+    matrix::SubTileSpec specHH() const noexcept {
+      return {{offset_, 0}, {nrows_, nrefls_}};
+    }
+
+    // Return the spec to use on the matrix HH are applied to
+    matrix::SubTileSpec specEV(const SizeType cols) const noexcept {
+      return {{origin_, 0}, {nrows_, cols}};
+    }
+
+  private:
+    SizeType origin_;
+    SizeType nrows_ = 0;
+    SizeType nrefls_ = 0;
+    SizeType offset_;
+  };
+
+  SizeType nrefls_;
+  matrix::SubTileSpec input_spec_;
+
+  bool acrossTiles_;
+  std::array<part_t, 2> parts_;
```

Comment:
If there are only two parts, why not have two `part_t` members with different names like `top/bottom` or `left/right` (not sure if these make sense)?

Reply:
Yes, it makes sense (top/bottom). I'm going to do it 😉

Reply:
See https://github.com/eth-cscs/DLA-Future/pull/481/commits/e2b97d04d8aaf2e4f973a0f58db8fd96e1132c2b